runtime/vm/extendedMessageNPE.cpp

```diff
@@ -20,21 +20,100 @@
  * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
  *******************************************************************************/
 
+#include "bcverify.h"
 #include "cfreader.h"
+#include "pcstack.h"
+#include "rommeth.h"
 #include "stackwalk.h"
-#include "util_api.h"
 #include "ut_j9vm.h"
+#include "vrfytbl.h"
 
 extern "C" {
 
+/* Define for debug
+#define DEBUG_BCV
+*/
+
+static IDATA buildBranchMap(J9NPEMessageData *npeMsgData);
+static void computeNPEMsgAtPC(J9VMThread *vmThread, J9ROMMethod *romMethod, J9ROMClass *romClass, UDATA npePC,
+		BOOLEAN npeFinalFlag, char **npeMsg, BOOLEAN *isMethodFlag, UDATA *temps, J9BytecodeOffset *bytecodeOffset);
 static char* convertToJavaFullyQualifiedName(J9VMThread *vmThread, J9UTF8 *fullyQualifiedNameUTF);
 static char* convertMethodSignature(J9VMThread *vmThread, J9UTF8 *methodSig);
+static IDATA copyToTargetStack(J9NPEMessageData *npeMsgData, UDATA target);
+static char* getCompleteNPEMessage(J9VMThread *vmThread, U_8 *bcCurrentPtr, J9ROMClass *romClass, char *npeCauseMsg, BOOLEAN isMethodFlag);
+static char* getFullyQualifiedMethodName(J9VMThread *vmThread, J9ROMClass *romClass, U_8 *bcIndex);
+static char* getLocalsName(J9VMThread *vmThread, J9ROMMethod *romMethod, U_16 localVar, UDATA bcCausePos, UDATA *temps);
+static char* getMsgWithAllocation(J9VMThread *vmThread, const char *msgTemplate, ...);
+static IDATA initializeNPEMessageData(J9NPEMessageData *npeMsgData);
+static void initStackFromMethodSignature(J9VMThread *vmThread, J9ROMMethod *romMethod, UDATA **stackTopPtr);
+static UDATA* pushViaSiganature(J9VMThread *vmThread, U_8 *signature, UDATA *stackTop, UDATA bcPos);
+static void setSrcBytecodeOffset(J9BytecodeOffset *bytecodeOffset, UDATA bcPos, UDATA first, UDATA second);
+static char* simulateStack(J9NPEMessageData *npeMsgData);
+
+#define BCV_INTERNAL_DEFAULT_SIZE (32*1024)
+#define BYTECODE_BRANCH_TARGET 0xFEFEFEFE
+#define BYTECODE_TEMP_CHANGED 0xFDFDFDFD
+#define BYTECODE_OFFSET_SHIFT 16
+#define BYTECODE_OFFSET_MASK 0xFFFF
+
+#define NPEMSG_FIRST_STACK() \
```

Comment:
Please put a blank line between the macros definitions.

Reply:
Fixed. Please have another look.